tests/sentry/integrations/slack/test_action_endpoint.py

```diff
@@ -398,28 +397,37 @@ def test_handle_submission_fail(self, client_put):
             associate_url=associate_url, user_email=self.user.email, org_name=self.org.name
         )
 
-    def test_invalid_token(self):
-        resp = self.post_webhook(token="invalid")
-        assert resp.status_code == 401
-
-    def test_no_integration(self):
+    @patch(
+        "sentry.integrations.slack.requests.SlackRequest._check_signing_secret", return_value=True
+    )
+    def test_no_integration(self, check_signing_secret_mock):
         self.integration.delete()
         resp = self.post_webhook()
         assert resp.status_code == 403
 
-    def test_slack_bad_payload(self):
+    @patch(
+        "sentry.integrations.slack.requests.SlackRequest._check_signing_secret", return_value=True
+    )
+    def test_slack_bad_payload(self, check_signing_secret_mock):
         resp = self.client.post("/extensions/slack/action/", data={"nopayload": 0})
         assert resp.status_code == 400
 
     def test_sentry_docs_link_clicked(self):
         payload = {
-            "token": options.get("slack.verification-token"),
             "team": {"id": "TXXXXXXX1", "domain": "example.com"},
             "user": {"id": self.identity.external_id, "domain": "example"},
             "type": "block_actions",
             "actions": [{"value": "sentry_docs_link_clicked"}],
         }
+
         payload = {"payload": json.dumps(payload)}
+        from tests.sentry.integrations.slack.test_requests import set_signature
+        from sentry import options
+        from urllib.parse import urlencode
+
+        headers = set_signature(
+            options.get("slack.signing-secret"), urlencode(payload).encode("utf-8")
+        )
 
-        resp = self.client.post("/extensions/slack/action/", data=payload)
+        resp = self.client.post("/extensions/slack/action/", data=payload, **headers)
```

Comment:
can't seem to get the signatures to match when I actually pass in the correct headers :(